src/main/java/filter/expression/Qualifier.java

```diff
@@ -214,16 +192,16 @@ public static void setCurrentTime(LocalDateTime dateTime) {
     }
 
     public boolean isEmptyQualifier() {
-        return name.isEmpty() && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.EMPTY;
     }
 
     public boolean isFalseQualifier() {
-        return name.equalsIgnoreCase("false") && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.FALSE;
```

Comment:
Ok I understand the first part.

On `Qualifier.EMPTY`, I think we can keep it. In many cases, we are returning `Qualifier` and not `QualifierType` - e.g. `map` function on `Negation`, `Conjunction` and `Disjunction`. If we remove `Qualifier.EMPTY`, then we need to construct it every time manually, which may introduce more bugs.